presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -369,4 +399,87 @@ public static LongWriteFunction timestampWriteFunction()
     }
+
+    public static Object jdbcObjectToPrestoNative(Object jdbcObject, Type prestoType)
+    {
+        if (jdbcObject == null) {
+            return null;
+        }
+        else if (prestoType instanceof DecimalType) {
+            DecimalType decimalType = (DecimalType) prestoType;
+            BigDecimal value = (BigDecimal) jdbcObject;
+            if (decimalType.isShort()) {
+                return encodeShortScaledValue(value, decimalType.getScale());
+            }
+            else {
+                return encodeScaledValue(value, decimalType.getScale());
+            }
+        }
+        else if (REAL.equals(prestoType)) {
+            return floatToRawIntBits((float) jdbcObject);
+        }
+        else if (jdbcObject instanceof Date) {
+            return toLong((Date) jdbcObject);
+        }
+        else if (jdbcObject instanceof Time) {
+            return toLong((Time) jdbcObject);
+        }
+        else if (jdbcObject instanceof Timestamp) {
+            return ((Timestamp) jdbcObject).getTime();
+        }
+        else {
+            return jdbcObject;
+        }
+    }
+
+    public static Object prestoNativeToJdbcObject(Type prestoType, Object prestoNative)
+    {
+        if (prestoNative == null) {
+            return null;
+        }
+        else if (prestoType instanceof DecimalType) {
+            DecimalType decimalType = (DecimalType) prestoType;
+            if (decimalType.isShort()) {
+                return toBigDecimal(decimalType, (long) prestoNative);
+            }
+            else {
+                return toBigDecimal(decimalType, (Slice) prestoNative);
+            }
+        }
+        else if (REAL.equals(prestoType)) {
+            return intBitsToFloat(toIntExact((long) prestoNative));
+        }
+        else if (TINYINT.equals(prestoType)) {
+            return SignedBytes.checkedCast((long) prestoNative);
+        }
+        else if (SMALLINT.equals(prestoType)) {
+            return Shorts.checkedCast((long) prestoNative);
+        }
+        else if (INTEGER.equals(prestoType)) {
+            return toIntExact((long) prestoNative);
+        }
+        else if (DATE.equals(prestoType)) {
+            return toDate((long) prestoNative);
+        }
+        else if (TIME.equals(prestoType)) {
+            return new Time((long) prestoNative);
+        }
+        else if (TIMESTAMP.equals(prestoType)) {
+            return new Timestamp((long) prestoNative);
+        }
+        else if (prestoNative instanceof Slice) {
+            if (prestoType instanceof VarcharType || prestoType instanceof CharType) {
+                return ((Slice) prestoNative).toStringUtf8();
+            }
+            else if (prestoType instanceof VarbinaryType) {
+                return ((Slice) prestoNative).getBytes();
+            }
+            else {
+                throw new PrestoException(FUNCTION_IMPLEMENTATION_ERROR, "Slice of unexpected type: " + prestoNative.getClass());
```

Comment:
- `prestoType` is what actually matters, so include that in the message (`prestoNative.getClass()` will be always `Slice` here)
- i am not sure `FUNCTION_IMPLEMENTATION_ERROR` is the right error code... if you can't find anything more proper, use generic err code instead